src/Components/HouseholdDataBlock/AgeInput.tsx

```diff
@@ -0,0 +1,137 @@
+import { FormControl, FormHelperText, InputLabel, MenuItem, Select } from '@mui/material';
+import { useEffect, useMemo } from 'react';
+import { FormattedMessage } from 'react-intl';
+import {
+  birthMonthErrorMessage,
+  birthMonthHasError,
+  birthMonthInvalidWithYearError,
+  birthMonthInvalidWithYearErrorMessage,
+  birthYearErrorMessage,
+  birthYearHasError,
+  useErrorController,
+} from '../../Assets/validationFunctions';
+import AutoComplete from '../AutoComplete/AutoComplete';
+import './AgeInput.css';
+
+const MONTHS = {
+  1: 'January',
+  2: 'Febuary',
+  3: 'March',
+  4: 'April',
+  5: 'May',
+  6: 'June',
+  7: 'July',
+  8: 'August',
+  9: 'September',
+  10: 'October',
+  11: 'November',
+  12: 'December',
+};
+
+const date = new Date();
+const CURRENT_YEAR = date.getFullYear();
+// January is 0 for getMonth
+const CURRENT_MONTH = date.getMonth() + 1;
+
+const MAX_AGE = 130;
+const YEARS = Array.from({ length: MAX_AGE }, (_, i) => {
+  const inputYear = CURRENT_YEAR - i;
+  return String(inputYear);
+});
+
+export function calcAge(birthYear: number, birthMonth: number) {
+  if (CURRENT_MONTH >= birthMonth) {
+    return CURRENT_YEAR - birthYear;
+  }
+
+  return CURRENT_YEAR - birthYear - 1;
+}
+
+type Props = {
+  birthMonth: number | null;
+  birthYear: number | null;
+  setBirthMonth: (month: number | null) => void;
+  setBirthYear: (year: number | null) => void;
+  submitted: number;
+};
+
+export default function AgeInput({ birthYear, birthMonth, setBirthMonth, setBirthYear, submitted }: Props) {
```

Comment:
What do you think about using `react-hook-form` and `zod` for these new components?

Reply:
I would have to refactor the whole question.